docs/usage/topology_aware_routing.md

```diff
@@ -0,0 +1,96 @@
+# Topology-Aware Traffic Routing
+
+The `TopologyAwareRouting` feature enables topology-aware traffic routing for components in the garden or seed clusters.
+
+## Motivation
+
+The enablement of [highly available Shoot control planes](shoot_high_availability.md) requires multi-zone Seed clusters. A Garden cluster can also be a multi-zone cluster. The topology-aware routing is introduced to reduce costs and to improve network performance for the cross availability zone traffic. The cross availability zone traffic is charged by the cloud providers and it comes with higher latency compared to the traffic within the same zone. The `TopologyAwareRouting` feature enables topology-aware routing for Services deployed in the garden or seed clusters. For the clients consuming these topology-aware Services, kube-proxy prefers the endpoints which are within the same zone.
```

Comment:
```suggestion
The enablement of [highly available shoot control-planes](shoot_high_availability.md) requires multi-zone seed clusters. A garden cluster can also be a multi-zone cluster. The topology-aware routing is introduced to reduce costs and to improve network performance for the cross availability zone traffic. The cross availability zone traffic is charged by the cloud providers and it comes with higher latency compared to the traffic within the same zone. The `TopologyAwareRouting` feature enables topology-aware routing for `Service`s deployed in the garden or seed clusters. For the clients consuming these topology-aware services, `kube-proxy` prefers the endpoints which are within the same zone.
```

Reply:
I cannot follow which one should be used consistently
"Shoot" vs "shoot"
"Seed" vs "seed"
"Garden cluster" vs "garden cluster"
"`Services`s" vs "Services" vs "services"
...

@n-boshnakov maybe you can write this down as general rule and make efforts that the rules are then enforced/applied.

Reply:
Achieving consistency for this is hard, and writing it down will probably not fix this problem since it depends on the context. To me, it depends on the context - when talking about a Kubernetes resource or object, I use `Shoot`(s), but when it's not directly related to such concrete resource then I use "shoot cluster". Not sure if this can be enforced correctly via some static checks.
Anyways, that's why we have such PR reviews. Will you make the change?

Reply:
I already made the change locally.
I am saying that we are missing guidelines for the above questions. See https://kubernetes.io/docs/contribute/style/style-guide/#use-upper-camel-case-for-api-objects. My understanding is that "Shoot" should be used over "shoot" no matter the context.

Reply:
🤷‍♂️

Reply:
As Rafael said, achieving consistency for the terminology across the whole repository will be hard, not to mention the websites where content coming from different repositories needs to be taken into account.

Currently I am working on proofreading all of the topics in the gardener/gardener repository, and trying to keep the terminology consistent on a per-topic basis. Achieving consistency across the whole repository would perhaps be a task for after that, combining in it changes to the [Contributors Guide](https://github.com/gardener/documentation/blob/master/website/documentation/contribute/_index.md) as well as the [Glossary](https://github.tools.sap/kubernetes/documentation/blob/master/website/documentation/090-resources/050-glossary/_index.md).